src/components/maps/MapListComponent.vue

```diff
@@ -0,0 +1,96 @@
+<template>
+    <div>
+        <div class="sort-options flex-row gap-md">
+            <SearchBox v-model="searchVal" />
+            <div class="sort-dropdown">
+                <Select v-model="sortMethod" :options="sortMethods" label="Sort By" />
+            </div>
+        </div>
+        <div class="map-list">
+            <div class="maps">
+                <MapOverviewCard v-for="(map, index) in filteredMaps" :key="index" :map="map" @map-selected="mapSelected" />
+            </div>
+        </div>
+    </div>
+</template>
+
+<script lang="ts" setup>
+/**
+ * TODO:
+ * - Similar to online map browser
+ * - Indicator whether map is installed or not
+ * - Easy one click install button
+ * - Demo map button that launches a simple offline game on the map
+ */
+
+import { computed, ref } from "vue";
+
+import SearchBox from "@/components/controls/SearchBox.vue";
+import Select from "@/components/controls/Select.vue";
+import MapOverviewCard from "@/components/maps/MapOverviewCard.vue";
+import { MapData } from "@/model/map-data";
+
+enum SortMethod {
+    Name = "Name",
+    Size = "Size",
+}
+
+const sortMethods = ref([SortMethod.Name, SortMethod.Size]);
+const sortMethod = ref(SortMethod.Name.valueOf());
+const searchVal = ref("");
```

Comment:
I'm leaning more towards banning enums entirely for this project in eslint, for reasons https://www.youtube.com/watch?v=0fTdCSH_QEU

Reply:
Replaced with union type